internal/app/init/internal/reg/reg.go

```diff
@@ -98,3 +99,12 @@ func (r *Registrator) ServiceList(ctx context.Context, in *empty.Empty) (result
 
 	return result, nil
 }
+
+// Stop implements the proto.InitServer interface and stops a
+// service running on Talos.
+func (r *Registrator) Stop(ctx context.Context, in *proto.StopRequest) (reply *proto.StopReply, err error) {
+	system.Services(r.Data).Stop(ctx, in.Id)
+
+	reply = &proto.StopReply{Resp: fmt.Sprintf("Service %s stopped", in.Id)}
```

Comment:
I might be purist here, but it might be worth returning an error from `Stop` if some id is not found? or if wait fails?

so that it doesn't say 'Service xyz stopped', while 'xyz' is not a service

Reply:
and might be that `%q` is better format than `%s` as it quotes output, so you can see if you have space or something embedded into the arg by mistake